src/os/FileStore.cc

```diff
@@ -1830,13 +1840,65 @@ void FileStore::queue_op(OpSequencer *osr, Op *o)
   dout(5) << "queue_op " << o << " seq " << o->op
 	  << " " << *osr
 	  << " " << o->bytes << " bytes"
-	  << "   (queue has " << throttle_ops.get_current() << " ops and " << throttle_bytes.get_current() << " bytes)"
+	  << "   (queue has " << throttle_ops.get_current() << " ops and " 
+          << throttle_bytes.get_current() << " bytes)"
 	  << dendl;
   op_wq.queue(osr);
+
+  if (g_conf->journal_dynamic_throttle) {
+  /* Trying to throttle the journal commit based on the number of
+     outstanding bytes to be applied. This will restrict the amount of
+     memory growth in the write path. The mode of throttle is to introduce
+     an incremental delay based on the % increase above an outstanding byte
+     threshold.
+   */
+    uint64_t outstanding_bytes = m_filestore_current_queue_bytes.read(); 
+    bool induce_delay = false;
+    uint32_t percentage_more = 0;
+    
+    if (outstanding_bytes > m_filestore_queue_max_bytes) {
+      percentage_more = (uint32_t)((100* (outstanding_bytes -
+                      m_filestore_queue_max_bytes))/m_filestore_queue_max_bytes);
+      if (percentage_more > 100) {
+        percentage_more = 100;
+      }
+
+      induce_delay = true;
+    }
+    
+    if (induce_delay) {
+
+      double delay_time;
+      dyn_throttle_filestore.calc_wait_time(100 - percentage_more, 
+                                                  delay_time, true);
+      if (0 != delay_time) {
+        struct timespec waittime;
+        waittime.tv_sec = 0;
+        waittime.tv_nsec = delay_time * 1000 * 1000 * 1000 ;
+        nanosleep(&waittime, NULL);
+        dout(5) << " Filestore on BaseDir = " << basedir << " and in journal on "
+            << journalpath << " is slow on applying transaction."
+            << " Throttling journal commits.Filestore op_wq depth = "
+            << op_queue.size() << " Outstanding bytes to be applied = "
+            << outstanding_bytes << " seq = " << o->op
+            << " Max op threshold = " << m_filestore_queue_max_ops
+            << " Max byte threshold = " << m_filestore_queue_max_bytes
+            << " Transaction size = " << o->bytes
+            << " Percentage exceeded threshold = " << percentage_more
+            << " Applying delay = " << delay_time << " in the journal commit path."
+            << dendl;
```

Comment:
Why not have both?  You explicitly disabled the normal throttles when yours is enabled.  I see no reason to do that.  If your throttling is working properly, it won't trigger anyway.